lib/iris/fileformats/pp.py

```diff
@@ -644,14 +644,30 @@ def ndim(self):
         return len(self.shape)
 
     def __getitem__(self, keys):
-        with open(self.path, 'rb') as pp_file:
-            pp_file.seek(self.offset, os.SEEK_SET)
-            data_bytes = pp_file.read(self.data_len)
-            data = _data_bytes_to_shaped_array(data_bytes,
-                                               self.lbpack,
-                                               self.boundary_packing,
-                                               self.shape, self.src_dtype,
-                                               self.mdi)
+        def is_emptyslice(key):
+            return (isinstance(key, slice) and
+                    isinstance(key.start, int) and
+                    key.start == key.stop)
+
+        if any(is_emptyslice(key) for key in keys):
```

Comment:
Assume you're comfortable with the scenario where only _some_ of the slices are empty, but we still avoid accessing the file at all?

Reply:
Absolutely, it seemed important to me to extend the capture beyond just exactly what dask does (at present).  In which case, it makes sense to me that it could sensibly skip some/any or all cases where no data is needed.  But exactly how to capture those I'm still thinking on (see [previous](https://github.com/SciTools/iris/pull/3659#discussion_r380237130))..